src/app/(proper_react)/(onboarding)/user/welcome/EnterInfo.tsx

```diff
@@ -4,22 +4,325 @@
+    { type: "dialog" },
+    explainerDialogState
+  );
+
+  const confirmDialogState = useOverlayTriggerState({});
+  const confirmDialogTrigger = useOverlayTrigger(
+    { type: "dialog" },
+    confirmDialogState
+  );
+
+  const l10n = useL10n();
+
+  const userDetailsData = [
+    {
+      label: l10n.getString("onboarding-enter-details-label-first-name"),
+      key: "firstName",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-first-name"
+      ),
+      value: firstName,
+      displayValue: firstName,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setFirstName,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-last-name"),
+      key: "lastName",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-last-name"
+      ),
+      value: lastName,
+      displayValue: lastName,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setLastName,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-location"),
+      key: "location",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-location"
+      ),
+      value: location,
+      displayValue: location,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setLocation,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-date-of-birth"),
+      key: "dateOfBirth",
+      type: "date",
+      placeholder: "",
+      value: dateOfBirth,
+      displayValue: new Date(dateOfBirth).toLocaleDateString("en-US"),
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setDateOfBirth,
+    },
+  ];
+
+  const validateEnteredInfo = () => {
+    const invalidInputKeys = userDetailsData
+      .filter(({ value }) => !getIsValidInfo(value))
+      .map(({ key }) => key);
+
+    setInvalidInputs(invalidInputKeys);
+  };
+
+  useEffect(() => {
+    if (!hasMounted.current) {
+      hasMounted.current = true;
+      return;
+    }
+
+    if (!invalidInputs?.length) {
+      confirmDialogState.open();
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [invalidInputs]);
+
+  const WhyDoWeNeedInfoDialog = () => (
+    <Dialog illustration={<Image src={whyWeNeedInfoHero} alt="" />}>
+      <div
+        className={`${viewStyles.dialogContents} ${enterInfoStyles.dialogContents}`}
+      >
+        <p>{l10n.getString("onboarding-enter-details-dialog")}</p>
+        <strong>
+          {l10n.getString("onboarding-enter-details-diaglog-subheader")}
+        </strong>
+
+        <ol className={viewStyles.list}>
+          <li className={viewStyles.listItem}>
+            <h4>
```

Comment:
Q: Not sure about the rest of the DOM outside this component, but do we have an H1-H3 before this beautiful `<h4>`?

Reply:
Good point! I just checked and we always have a `h1` but not necessarily a `h2`. Dialog titles are a `h3`. Since we can’t guarantee a `h2` I’d need to change the Dialog component. I’d prefer to not add more scope to this PR so  I’m leaning towards removing the header elements and reviewing the header hierarchy as soon as the entire “welcome” flow is in place.